bundles/config/org.eclipse.smarthome.config.core/src/main/java/org/eclipse/smarthome/config/core/internal/i18n/I18nConfigOptionsProvider.java

```diff
@@ -23,29 +29,65 @@
  * {@link ConfigOptionProvider} that provides a list of
  *
  * @author Simon Kaufmann - initial contribution and API
+ * @author Erdoan Hadzhiyusein - Added time zone
  *
  */
 @Component(immediate = true)
 public class I18nConfigOptionsProvider implements ConfigOptionProvider {
 
+    private static final String NO_OFFSET_FORMAT = "(GMT) %s";
+    private static final String NEGATIVE_OFFSET_FORMAT = "%s (GMT%d:%02d)";
+    private static final String POSITIVE_OFFSET_FORMAT = "%s (GMT+%d:%02d)";
+
     @Override
     public Collection<ParameterOption> getParameterOptions(URI uri, String param, Locale locale) {
         if (uri.toString().equals("system:i18n")) {
             Locale translation = locale != null ? locale : Locale.getDefault();
-            if (param.equals("language")) {
-                return getAvailable(locale, l -> new ParameterOption(l.getLanguage(), l.getDisplayLanguage(translation)));
-            } else if (param.equals("region")) {
-                return getAvailable(locale, l -> new ParameterOption(l.getCountry(), l.getDisplayCountry(translation)));
-            } else if (param.equals("variant")) {
-                return getAvailable(locale, l -> new ParameterOption(l.getVariant(), l.getDisplayVariant(translation)));
+            switch (param) {
+                case "language":
+                    return getAvailable(locale,
+                            l -> new ParameterOption(l.getLanguage(), l.getDisplayLanguage(translation)));
+                case "region":
+                    return getAvailable(locale,
+                            l -> new ParameterOption(l.getCountry(), l.getDisplayCountry(translation)));
+                case "variant":
+                    return getAvailable(locale,
+                            l -> new ParameterOption(l.getVariant(), l.getDisplayVariant(translation)));
+                case "timezone":
+                    Collection<ParameterOption> collection = new LinkedList<ParameterOption>();
```

Comment:
please factor out this logic into its own method so it stays easy to keep an overview of this switch statement (good idea, by the way!). 

And while you are at it: why didn't you keep using the stream? e.g. like 

```java
        return ZoneId.getAvailableZoneIds().stream().map(TimeZone::getTimeZone).sorted((tz1, tz2) -> {
            return tz2.getRawOffset() - tz2.getRawOffset();
        }).map(tz -> {
            return new ParameterOption(tz.getID(), getTimeZoneRepresentation(tz));
        }).collect(Collectors.toList());

```